lib/trino-orc/src/main/java/io/trino/orc/OrcReader.java

```diff
@@ -101,8 +101,6 @@ private static Optional<OrcReader> createOrcReader(
             Optional<OrcWriteValidation> writeValidation)
             throws IOException
     {
-        orcDataSource = wrapWithCacheIfTiny(orcDataSource, options.getTinyStripeThreshold());
```

Comment:
I believe the point of this optimization is to reduce request count on filesystem. You always need to make at least one filesystem request to read file footer, that's why we're reading entire file here when the file is small. If you delay this until later then we make 2 filesystem requests instead of one when the file is small.
The test failures in this PR are also because of that. We most likely don't want to make this change.
If you're more concerned about the size of read rather than the request count, you can tune `hive.orc.tiny-stripe-threshold` lower (default is 8MB).
cc: @dain